solo-chains/runtime/dancelight/src/lib.rs

```diff
@@ -915,14 +984,22 @@ impl ProcessMessage for MessageProcessor {
         meter: &mut WeightMeter,
         id: &mut [u8; 32],
     ) -> Result<bool, ProcessMessageError> {
-        let para = match origin {
-            AggregateMessageOrigin::Ump(UmpQueueId::Para(para)) => para,
-        };
-        xcm_builder::ProcessXcmMessage::<
-            Junction,
-            xcm_executor::XcmExecutor<xcm_config::XcmConfig>,
-            RuntimeCall,
-        >::process_message(message, Junction::Parachain(para.into()), meter, id)
+        match origin {
+            AggregateMessageOrigin::Ump(UmpQueueId::Para(para)) => {
+                xcm_builder::ProcessXcmMessage::<
+                    Junction,
+                    xcm_executor::XcmExecutor<xcm_config::XcmConfig>,
+                    RuntimeCall,
+                >::process_message(
+                    message, Junction::Parachain(para.into()), meter, id
+                )
+            }
+            AggregateMessageOrigin::Snowbridge(_) => {
+                tp_bridge::CustomProcessSnowbridgeMessage::<Runtime>::process_message(
```

Comment:
we could have snowbridge follow the previous path (through ethereumoutboundqueue) and come up with a new origin that routes everything through our `customProcessSnowbridgeMessage`... this origin is set in the sendMessage implementation of `EthereumOutboundQueue`. The solution to that would be to implement a differnt `SendMessage` implementation that sets it to the proper origin, and use this on external-validator-slashes

Reply:
Done, there are 2 different origins: Snowbridge for messages that are compatible with Snowbridge, and SnowbridgeTanssi for the Tanssi messages exclusively. Better name ideas welcome.